openml/datasets/dataset.py

```diff
@@ -173,6 +173,28 @@ def __init__(self, name, description, format=None,
         else:
             self.data_pickle_file = None
 
+    def __str__(self):
+        object_dict = self.__dict__
+        output_str = ''
+        name = '\n%-14s: %s\n' % ("Name", object_dict['name'])
```

Comment:
I don't think it is necessary to add the identifier via string interpolation, you can simply have it in the main string.